### PR TITLE
I18n refactor

### PR DESCRIPTION
**Pull request by ouranos**

### CODE REVIEW

**Comment by ouranos** [comment]
@alexnoox I'll need to add a couple more commits later today, but could you have a quick look first?

**Comment by ouranos** [comment]
@alexnoox I've just pushed the updated version.
I'm pulling my hairs on the spec as everything is called on the index.run. I've parked it for now.

One issue is that we won't have the impac locales in dev unless you run a rake task. We can have a chat  on the best way to tackle that.

File: src/app/components/mno-locale-config/mno-locale-config.svc.coffee [hunk 1]
```diff
@@ -0,0 +1,74 @@
+angular.module 'mnoEnterpriseAngular'
+  # This service is used to manage the configuration of $translate
+  .service('MnoLocaleConfigSvc', (
+    $q, $window, $translate,
+    MnoeApiSvc, MnoeFullApiSvc, MnoeCurrentUser,
+    I18N_CONFIG, LOCALES, URI
```
Comment (alexnoox) [inline src/app/components/mno-locale-config/mno-locale-config.svc.coffee]: There is not too many services, you can write it on one line.

File: src/app/components/mno-locale-config/mno-locale-config.svc.coffee [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      # Extract the language code if present
+      re = /^\/([A-Za-z]{2}(-[A-Z]{2})?)\/dashboard\//i
+      found = path.match(re)
+
+      # Ex found: ["/en/dashboard/", "en", index: 0, input: "/en/dashboard/"]
+      locale = found[1] if found?
+
+      $q.resolve(locale)
+
+
```
Comment (alexnoox) [inline src/app/components/mno-locale-config/mno-locale-config.svc.coffee]: Remove empty line

File: src/app/views/layout.controller.coffee [hunk 3]
```diff
@@ -1,5 +1,5 @@
 angular.module 'mnoEnterpriseAngular'
-  .controller 'LayoutController', ($scope, $stateParams, $state, $q, MnoeCurrentUser, MnoeOrganizations) ->
+  .controller 'LayoutController', ($scope, $stateParams, $state, $q, MnoeCurrentUser, MnoeOrganizations, $translate, $rootScope) ->
```
Comment (alexnoox) [inline src/app/views/layout.controller.coffee]: DI order: native services > lib services > app services > contants

Comment (ouranos) [inline src/app/views/layout.controller.coffee]: My bad these services are no longer needed, they were used for debugging

Comment (ouranos) [inline src/app/views/layout.controller.coffee]: Actually this code doesn't seem to be needed anymore

**Comment by ouranos** [comment]
@alexnoox please review the new commits.

File: gulp_tasks/locales.js [hunk 4]
```diff
@@ -1,12 +1,50 @@
...
+const merge = require('merge-stream');
+const merge_json = require('gulp-merge-json');
 
-gulp.task('locales', locales);
+const conf = require('../conf/gulp.conf');
+const localesPath = 'bower_components/impac-angular/dist/locales/';
+
+// Return a list of locales file from a folder
+function getLocales(dir) {
+  return fs.readdirSync(dir)
```
Comment (alexnoox) [inline gulp_tasks/locales.js]: Can't you do something like `return fs.readdirSync(path.join(dir + "/*.json")`

File: gulp_tasks/locales.js [hunk 5]
```diff
@@ -1,12 +1,50 @@
...
+    .filter(function(file) {
+      return file.substr(-5) === '.json';
+    });
+}
+
+// Append their locale to our locale
+function mergeLocales(our, their) {
```
Comment (alexnoox) [inline gulp_tasks/locales.js]: `main, remote`? `orig, remote`?

File: gulp_tasks/locales.js [hunk 6]
```diff
@@ -1,12 +1,50 @@
...
 function locales() {
-  return gulp.src('bower_components/impac-angular/dist/locales/*')
-    .pipe(rename(function (filename) {
-      filename.basename = filename.basename.slice(0, 2);
-    }))
-    .pipe(gulp.dest('src/locales/impac'))
+  var locales = getLocales(localesPath);
+
+  // Executes the function once per file and returns the async stream
+  // This will append 'en-AU.json' to 'en-AU.json'
```
Comment (alexnoox) [inline gulp_tasks/locales.js]: `// This will append 'some-dir/en-AU.json' to 'src/locales/en-AU.json'`

File: gulp_tasks/locales.js (hunk 4 above)
Comment (ouranos) [reply gulp_tasks/locales.js]: Nope, but I've solved it using `glob.sync`.

File: gulp_tasks/locales.js (hunk 5 above)
Comment (ouranos) [reply gulp_tasks/locales.js]: Done, clearer indeed

File: gulp_tasks/locales.js (hunk 6 above)
Comment (ouranos) [reply gulp_tasks/locales.js]: Done

**Comment by ouranos** [comment]
@alexnoox ready for last review.
Let me know and I'll rebase before merge. Thanks

**Pull request merged**